prism_compile.c

```diff
@@ -755,6 +755,11 @@ pm_static_literal_value(rb_iseq_t *iseq, const pm_node_t *node, const pm_scope_n
       case PM_HASH_NODE: {
         const pm_hash_node_t *cast = (const pm_hash_node_t *) node;
         const pm_node_list_t *elements = &cast->elements;
+        // If there are 0 elements, we have an empty array, so we want to
+        // return nil as the value in order to use the newhash instruction.
+        if (elements->size == 0) {
+            return Qnil;
+        }
```

Comment:
I'm definitely not feeling confident this is the right way to handle this. But if we have no elements there's no reason to go through the rest, we know it's an empty hash and should use `newhash` instead of `duphash`.

Reply:
Yeah, I'd be nervous about this, because it's not really returning the value that it represents. Instead, can you check the number of elements where we compile `PM_HASH_NODE` and call into this function?

Reply:
Ah ok, yea I can do that!